### DISCUSSION TITLE
How do you get the long text files to work.

### DISCUSSION THREAD

**Discussion by Matthomvalle** [neonbjb/tortoise-tts#485] [Q&A]
So I'm not really sure what I'm doing wrong. I'm able to use 
python tortoise/do_tts.py --text "I'm going to speak this" --voice random --preset fast
but when it comes to the textfile command it never works for me I have altered it in many ways trying to tweak it. At first I thought it was a directory problem. I made a directory called. Text on my main Tortoise TTS files and place the txt file there and named it supertest.txt

Just don't know what to try anymore.
This is the ENVS set-up that I used

conda create --name tortoise-tts python=3.9

conda activate tortoise-tts

cd anaconda3

 conda install torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia

conda install -c conda-forge pysoundfile

git clone https://github.com/neonbjb/tortoise-tts.git

cd tortoise-tts

EDIT requirement.txt FILE Location 
C:\Users\YOUR USERNAME\anaconda3\tortoise-tts
open it with notepad
delete numba and lvm then edit numpy and add this version 1.25 and for scipy this version 1.10. then continue with the command below.

python -m pip install -r ./requirements.txt

conda install numba

![Tortoise-tts directory](https://github.com/neonbjb/tortoise-tts/assets/135963847/07905647-5877-4f0c-b56b-688cd11da106)
![Anaconda prompt error](https://github.com/neonbjb/tortoise-tts/assets/135963847/f2dd17a3-56f1-4d8c-88d0-79fa5dda44d0)

**Comment by dekker33** [comment]
I'm having the same issue. Can someone help with this?

I have do-tts.py working.

When I try the read.py I get directory errors.

I've tried multiple variations of the prompt

1. python tortoise/read.py --textfile <the virus is from china> --voice=<trump>
2. python tortoise/read.py --textfile <trump.txt> --voice=<trump>                                                  (with a text file named trump in /data)
3. python tortoise/read.py --textfile trump.txt --voice trump   
4. python tortoise/read.py --textfile "the virus is from china" --voice=<trump>
5. python tortoise/read.py --textfile "the virus is from china" --voice trump

I can't figure it out. Can someone give me an example of EXACTLY how to write this prompt?

**Comment by neonbjb** [comment]
The text file you are reading from is encoded in something other than
UTF-8. Try re-encoding it (chatGPT can help!) and re-running the script.
You could also modify read.py to fix this problem inline, consider
submitting a PR if you do.

On Mon, Jun 26, 2023 at 9:11 AM dekker33 ***@***.***> wrote:

> I'm having the same issue. Can someone help with this?
>
> I have do-tts.py working.
>
> When I try the read.py I get directory errors.
>
> I've tried multiple variations of the prompt
>
>    1. python tortoise/read.py --textfile --voice=
>    2. python tortoise/read.py --textfile <trump.txt> --voice= (with a
>    text file named trump in /data)
>    3. python tortoise/read.py --textfile trump.txt --voice trump
>    4. python tortoise/read.py --textfile "the virus is from china"
>    --voice=
>    5. python tortoise/read.py --textfile "the virus is from china"
>    --voice trump
>
> I can't figure it out. Can someone give me an example of EXACTLY how to
> write this prompt?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/neonbjb/tortoise-tts/discussions/485#discussioncomment-6283351>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGLMOQSEFCKYCD5QATMCATXNGRCNANCNFSM6AAAAAAZLMCW44>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
-
James Betker

**Reply by dekker33** [reply]
Hey neon, thanks for the help.

I think it's working!

I changed the .txt encoding to "ANSI"

AND

I normalized my audio, re-rendered it at 32-bit floating point.

It was 64 bit floating point before.

Thanks so much man.
